indico/modules/events/papers/models/papers.py

```diff
@@ -66,3 +68,45 @@ def state(self):
     @state.setter
     def state(self, state):
         self.last_revision.state = state
+
+    def is_in_final_state(self):
+        return self.state in {PaperRevisionState.accepted or PaperRevisionState.rejected}
+
+    def can_comment(self, user):
+        if not user:
+            return False
+        return self.is_submitter(user) or self.can_judge(user) or self.can_review(user)
+
+    def is_submitter(self, user):
+        if not user:
+            return False
+        return user in {r.submitter for r in self.revisions} or user in self.contribution.submitters
```

Comment:
Yes, I would allow all 'persons' associated with the contribution to upload papers. After all, the authors are the ones most likely to also write and upload the paper